cmd/snap/cmd_whoami_test.go

```diff
@@ -0,0 +1,54 @@
+package main_test
+
+import (
+	"net/http"
+
+	"github.com/snapcore/snapd/osutil"
+
+	. "gopkg.in/check.v1"
+
+	snap "github.com/snapcore/snapd/cmd/snap"
+)
+
+func (s *SnapSuite) TestWhoamiLoggedInUser(c *C) {
+	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
+		c.Errorf("whoami does not hit the API, it just reads ~/.snap/auth.json (which is mocked in the tests)")
+	})
+
+	s.Login(c)
+
+	_, err := snap.Parser(snap.Client()).ParseArgs([]string{"whoami"})
+	c.Assert(err, IsNil)
+	c.Check(s.Stdout(), Equals, "email: hello@mail.com\n")
+
+	s.Logout(c)
```

Comment:
It is unclear what this logout is for. If it is to clean up some state it should be paired with `s.Login` call above in the style of `defer s.Logout(c)`